src/plugins/intel_gpu/src/graph/permute.cpp

```diff
@@ -137,7 +137,9 @@ void permute_inst::update_output_memory() {
     if (_node != nullptr)
         build_deps();
 
-    _outputs = {_network.get_engine().reinterpret_buffer(input_memory(), _impl_params->get_output_layout())};
+    GPU_DEBUG_TRACE_DETAIL << id() << " : update_output_memory with mem of input " << get_node().get_dependency(0).id()
+                           << " : " << input_memory_ptr()->buffer_ptr() << std::endl;
+    _outputs[0] = input_memory_ptr();
```

Comment:
Why not reinterpret?

Reply:
Reverted. Previously, `update_output_memory` was also called when creating `primitive_inst`. so there was an issue of retrieving the output layout even in the case where impl_params is dynamic. I fixed this corner case itself.